### PR TITLE
Renamer: Warn about out of scope identifiers.

### PR DESCRIPTION
**Pull request by sjakobi**
Inspired by https://github.com/haskell/haddock/issues/817.

### CODE REVIEW

**Comment by alexbiehl** [comment]
Looks good. IIRC @hvr wants a warning for ambiguous identifiers. This might be the perfect timing.

File: haddock-api/src/Haddock/Interface/LexParseRn.hs [hunk 1]
```diff
@@ -113,7 +115,7 @@ rename dflags gre = rn
               -- type constructor names (such as in #253). So now we
               -- only get type constructor links if they are actually
               -- in scope.
-              a:_ -> pure (outOfScope dflags a)
+              a:_ -> outOfScope dflags a
```
Comment (sjakobi) [inline haddock-api/src/Haddock/Interface/LexParseRn.hs]: > IIRC @hvr wants a warning for ambiguous identifiers.

You mean this case…

**Comment by alexbiehl** [comment]
Correct

Simon Jakobi <notifications@github.com> schrieb am Di., 8. Mai 2018, 19:04:

**Comment by sjakobi** [comment]
What are users supposed to do about such a warning? I'm not aware of any syntax to disambiguate data constructors and type constructors.

**Comment by alexbiehl** [comment]
Yeah that would need to be invented :) there is a ticket somewhere.. on
mobile right now.

Simon Jakobi <notifications@github.com> schrieb am Di., 8. Mai 2018, 19:18:

**Pull request merged**